iommi/fragment.py

```diff
@@ -187,7 +187,9 @@ def render_text_or_children(self, context):
         )
 
     def __repr__(self):
-        if self.is_refine_done:
+        if self._is_bound:
```

Comment:
the bound case is obsolete with your fix

Reply:
True. I'll do a rebase and cleanup from the leftovers